test/filtering.html

```diff
@@ -181,6 +181,13 @@
         flushGrid(grid);
       });
 
+      it('should not throw when receiving null filter', done => {
+        grid._filters[0].value = null;
```

Comment:
Add `flushFilters(grid);` right after this line to invalidate grid filters. Without `flushFilters(grid);` test is passing without your fix in `vaadin-grid-array-data-provider-mixin.html`